### PR TITLE
Added configurable serviceType and ingress

### PR DESCRIPTION
**Pull request by jonaseck2**
* Bump version
* Increased template field truncation length
* Added configuratble servicetype and ingress

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

**Comment by jonaseck2** [comment]
/assign @seanknox

Den 10 jan. 2018 11:59 skrev "k8s-ci-robot" <notifications@github.com>:

Hi @jonaseck2 <https://github.com/jonaseck2>. Thanks for your PR.

I'm waiting for a kubernetes <https://github.com/orgs/kubernetes/people>
member to verify that this patch is reasonable to test. If it is, they
should reply with /ok-to-test on its own line. Until that is done, I will
not automatically test new commits in this PR, but the usual testing
commands by org members will still work. Regular contributors should join
the org <https://git.k8s.io/community/community-membership.md#member> to
skip this step.

I understand the commands that are listed here
<https://github.com/kubernetes/test-infra/blob/master/commands.md>.

Instructions for interacting with me using PR comments are available here
<https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md>.
If you have questions or suggestions related to my behavior, please file an
issue against the kubernetes/test-infra
<https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
repository.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/charts/pull/3284#issuecomment-356569384>,
or mute
the thread

File: stable/spark/templates/spark-master-deployment.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ spec:
       targetPort: {{ .Values.WebUi.ContainerPort }}
   selector:
     component: "{{ .Release.Name }}-{{ .Values.Master.Component }}"
-  type: "LoadBalancer"
+  type: {{ .Values.Master.ServiceType | default "LoadBalancer" }}
```
Comment (unguiculus) [inline stable/spark/templates/spark-master-deployment.yaml]: Add the default to `values.yaml` instead of using the `default` function.

**Review by unguiculus — DISMISSED** [review]
Thanks for the PR.

File: stable/spark/templates/spark-master-deployment.yaml (hunk 1 above)
Comment (jonaseck2) [reply stable/spark/templates/spark-master-deployment.yaml]: Done

**Comment by unguiculus** [comment]
Needs a version bump.

**Comment by lachie83** [comment]
@jonaseck2 could you please do another version bump

**Comment by jonaseck2** [comment]
Accidentally bumped #3291 instead, oh well :)

**Pull request merged**